src/framework/testing.ts

```diff
@@ -0,0 +1,66 @@
+import getPort from 'get-port'
+import { GraphQLClient } from 'graphql-request'
+import { App } from './app'
+import * as Layout from './layout'
+
+type AppClient = {
+  query: GraphQLClient['request']
+}
+
+export function createQueryClient(apiUrl: string): AppClient {
+  const client = new GraphQLClient(apiUrl)
+
+  return {
+    query(queryString, variables) {
+      return client.request(queryString, variables)
+    },
+  }
+}
+
+export type TestContext = {
+  app: AppClient
+  teardown: () => Promise<void>
+}
+
+/**
+ * Setup a test context providing utilities to query against your GraphQL API
+ */
+export async function setupTest(): Promise<TestContext> {
+  process.env.GRAPHQL_SANTA_STAGE = 'dev'
+
+  const port = await getPort({ port: getPort.makeRange(4000, 6000) })
+  const apiUrl = `http://localhost:${port}/graphql`
+
+  const app = require('./index').app as App
```

Comment:
Why dynamic/lazy?

Reply:
I figured it'd be safer to trigger the singleton instantiation only when calling the function and not when importing /testing. Not sure it has any benefits, can revert that

Reply:
Yeah I see what you mean. Still, hmmm... let's go optimistically on this (`import`) until we hit problems.